### PR TITLE
feat: simplify prompts with metadata prompt

### LINKED ISSUE #458: rework mr creation to more closely assemble gh
**Describe the feature or problem you'd like to solve**

I think the pr creation process of `gh` is much simpler:

- It doesn't prompt for `milestones`, `labels` and `assignees`, instead at the end you can `Add metadata`

**Propose a Solution**

Add a `Add metadata` prompt, followed by allowing the user to multiple choice pick either of the 3 attributes listed above, then present the prompts for the ones he picked.

### PR DESCRIPTION
**Pull request by maxice8**
**Description**

- feat(commands/cmdutils): add option to 'Add Metadata'
- feat(commands/cmdutils): add variables to represent actions
- refactor(commands/cmdutils): don't prompt user for confirmation
- feat(commands/cmdutils): Add function to prompt types of metadata to add
- feat(commands/issue/create): implement 'Add metadata' prompt

**Related Issue**

Resolves #458

**How Has This Been Tested?**

https://gitlab.alpinelinux.org/alpine/aports/-/issues/12260, see screeenshots

**Screenshots (if appropriate):**
interactive-add-metadata-1
interactive-add-metadata-2
interactive-add-metadata-3
interactive-add-metadata-4
interactive-add-metadata-5

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

### CODE REVIEW

**Comment by maxice8** [comment]
@profclems This is a `Draft` because I didn't implement it yet for `mr create` which causes build failures due to change in signatures of functions used.

I just want to get your input if it is a idea worth pursuing now that I have a working **prototype**

**Comment by profclems** [comment]
I'm very much excited about this one. 
This is much simpler and avoids prompting the user so many times

**Comment by maxice8** [comment]
It is no longer draft and I have it implemented for `mr create`, there might be worth in the future to pursue a interactive editor alternative for the `update` subcommands of `issue` and `mr`.

Something like `glab issue edit` that will open a metadata prompt for all 5 metadata types and even more.

**Comment by profclems** [comment]
I feel we have to change the assignees prompt label to read `Assignees by usernames [comma separated]: ` instead of just `username(s) [comma separated]`

> Something like `glab issue edit` that will open a metadata prompt for all 5 metadata types and even more.

Sounds good for the future.

There are some discussions going on within GitLab whether to adopt this tool and make it their official CLI together with another tool called lab which also works like GitHub's [hub](https://github.com/github/hub)

https://gitlab.com/groups/gitlab-org/-/epics/3324

File: commands/cmdutils/cmdutils.go [hunk 1]
```diff
@@ -132,108 +132,90 @@ func DescriptionPrompt(response *string, templateContent, editorCommand string)
...
+	*response = milestoneMap[selectedMilestone].ID
+
 	return nil
 }
 
 func AssigneesPrompt(response *[]string) (err error) {
-	var addAssignees bool
-	err = prompt.Confirm(&addAssignees, "Do you wish to assign users?", true)
+	var responseString string
+	err = prompt.AskQuestionWithInput(&responseString, "Username(s) [Comma Separated]", "", false)
```
Comment (profclems) [inline commands/cmdutils/cmdutils.go]: To eliminate ambiguity:
```suggestion
	err = prompt.AskQuestionWithInput(&responseString, "Assignee username(s) [Comma Separated]", "", false)
```

Comment (maxice8) [reply commands/cmdutils/cmdutils.go]: fixed

**Review by profclems — APPROVED** [review]
All looks good. I'm happy to merge this 🎉

**Pull request merged**